packages/website/src/components/editor/LoadedEditor.tsx

```diff
@@ -83,8 +83,28 @@ export const LoadedEditor: React.FC<LoadedEditorProps> = ({
     const markers = monaco.editor.getModelMarkers({
       resource: model.uri,
     });
-    onMarkersChange(parseMarkers(markers, codeActions, editor));
-  }, [codeActions, onMarkersChange, editor, monaco.editor]);
+
+    const errors = parseMarkers(markers, codeActions, editor);
+
+    onMarkersChange(prev => {
+      const tsconfigError =
+        activeTab === 'tsconfig' &&
+        Object.fromEntries(prev[activeTab].map(error => [error.group, error]))
+          .Typescript;
```

Comment:
[Refactor] This confused me at first - we don't really need a whole grouped object, just the first tsconfig error.

```suggestion
        Object.values(prev[activeTab]).find(
          error => error.group === 'TypeScript',
        );
```

In fact, we don't care about this at all if `errors.length`, so this callback body can be rewritten like:

```ts
const tsconfigErrors =
  activeTab === 'tsconfig' &&
  !errors.length &&
  Object.values(prev[activeTab]).filter(
    error => error.group === 'TypeScript',
  );

return {
  ...prev,
  [activeTab]: tsconfigErrors || errors,
};
```

Reply:
The code you suggested is much clearer, thank you